packages/react/src/use-chat.ts

```diff
@@ -714,3 +694,46 @@ function countTrailingAssistantMessages(messages: Message[]) {
   }
   return count;
 }
+
+async function prepareAttachmentsForRequest(
+  attachmentsFromOptions: FileList | Array<Attachment> | undefined,
+) {
+  const draftAttachmentsForRequest: Array<Attachment> = [];
+
+  if (attachmentsFromOptions) {
+    if (attachmentsFromOptions instanceof FileList) {
+      for (const attachment of Array.from(attachmentsFromOptions)) {
+        const { name, type } = attachment;
+
+        const dataUrl = await new Promise<string>((resolve, reject) => {
+          const reader = new FileReader();
+          reader.onload = readerEvent => {
+            resolve(readerEvent.target?.result as string);
+          };
+          reader.onerror = error => reject(error);
+          reader.readAsDataURL(attachment);
+        });
+
+        draftAttachmentsForRequest.push({
+          name,
+          contentType: type,
+          url: dataUrl,
+        });
+      }
+    } else if (Array.isArray(attachmentsFromOptions)) {
+      for (const file of attachmentsFromOptions) {
+        const { name, url, contentType } = file;
+
+        draftAttachmentsForRequest.push({
+          name,
+          contentType,
+          url,
+        });
+      }
+    } else {
+      throw new Error('Invalid attachments type');
+    }
+  }
+
+  return draftAttachmentsForRequest;
```

Comment:
now that it's a separate method, i would recommend using the guard pattern to simplify the structure.

https://deviq.com/design-patterns/guard-clause

i.e.
 ```ts
if (attachmentsFromOptions == null) {
  return []
}
```
as the first statement to reduce the level of indentation.

You can then also use `map` and directly return the result, eliminating the `draftAttachmentsFromRequest` variable.

Reply:
e.g.

```ts
async function prepareAttachmentsForRequest(
  attachmentsFromOptions: FileList | Array<Attachment> | undefined
) {
  if (!attachmentsFromOptions == null) {
    return []
  }

  if (attachmentsFromOptions) {
    const draftAttachmentsForRequest: Array<Attachment> = []
    if (attachmentsFromOptions instanceof FileList) {
      for (const attachment of Array.from(attachmentsFromOptions)) {
        const { name, type } = attachment

        const dataUrl = await new Promise<string>((resolve, reject) => {
          const reader = new FileReader()
          reader.onload = readerEvent => {
            resolve(readerEvent.target?.result as string)
          }
          reader.onerror = error => reject(error)
          reader.readAsDataURL(attachment)
        })

        draftAttachmentsForRequest.push({
          name,
          contentType: type,
          url: dataUrl
        })
      }
    }
    return draftAttachmentsForRequest
  }

  if (Array.isArray(attachmentsFromOptions)) {
    return attachmentsFromOptions
  }

  throw new Error('Invalid attachments type')
}
```

Reply:
(the above reduces the cognitive complexity and makes it clear that the FileList handling is in fact the trickier part)

Reply:
oh til! I definitely like the clean up – will update and keep this pattern in mind moving forward!